src/unitxt/operators.py

```diff
@@ -971,7 +971,22 @@ def process(self, stream: Stream, stream_name: Optional[str] = None) -> Generato
                 yield instance
 
 
-class ExtractFieldValues(MultiStreamOperator):
+class AddFieldValues(SingleStreamOperator):
```

Comment:
We already have a class AddFields(StreamInstanceOperator), why do we need this one?

Reply:
I am aware of this `AddFields`. But, could very well be due to my ignorance, I needed a Stream processor, and not an instance processor. I looked for an example with generators and with streams, and found `SplitByValue`, and adapted from it. Perhaps I can learn something now.  E.g. if I knew how to produce a StreamProcessor  (whose .process() is fed with stream, and it outputs a generator of a stream) from a StreamInstance processor which like `AddFields`, simply returns a dict.